diff --git a/arch/mips/Kbuild b/arch/mips/Kbuild
index d2cfe45..ee70921 100644
--- a/arch/mips/Kbuild
+++ b/arch/mips/Kbuild
@@ -1,7 +1,7 @@
 # Fail on warnings - also for files referenced in subdirs
 # -Werror can be disabled for specific files using:
 # CFLAGS_<file.o> := -Wno-error
-subdir-ccflags-y := -Werror
+#subdir-ccflags-y := -Werror
 
 # platform specific definitions
 include arch/mips/Kbuild.platforms
diff --git a/arch/mips/xburst/soc-x1000/common/reset.c b/arch/mips/xburst/soc-x1000/common/reset.c
old mode 100644
new mode 100755
index 9477b37..5718985
--- a/arch/mips/xburst/soc-x1000/common/reset.c
+++ b/arch/mips/xburst/soc-x1000/common/reset.c
@@ -106,6 +106,8 @@ int inline reset_keep_power(void)
 			     inl(RTC_IOBASE + RTC_PWRONCR) & ~(1 << 0));
 }
 
+EXPORT_SYMBOL(reset_keep_power);
+
 #define HWFCR_WAIT_TIME(x) ((x > 0x7fff ? 0x7fff: (0x7ff*(x)) / 2000) << 5)
 #define HRCR_WAIT_TIME(x) ((((x) > 1875 ? 1875: (x)) / 125) << 11)
 
diff --git a/include/linux/compiler-gcc.h b/include/linux/compiler-gcc.h
index 24545cd..4de3906 100644
--- a/include/linux/compiler-gcc.h
+++ b/include/linux/compiler-gcc.h
@@ -100,7 +100,13 @@
 #define __gcc_header(x) #x
 #define _gcc_header(x) __gcc_header(linux/compiler-gcc##x.h)
 #define gcc_header(x) _gcc_header(x)
-#include gcc_header(__GNUC__)
+//#include gcc_header(__GNUC__)
+#if __GNUC__ < 5 
+#define __LINUX_GCC_VERSION __GNUC__ 
+#else 
+#define __LINUX_GCC_VERSION 4 
+#endif 
+#include gcc_header(__LINUX_GCC_VERSION)
 
 #if !defined(__noclone)
 #define __noclone	/* not needed */
